app/controllers/AnnotationIOController.scala

```diff
@@ -198,19 +201,31 @@ Expects:
 
   private def findDataSetForUploadedAnnotations(
       skeletonTracings: List[SkeletonTracing],
-      volumeTracings: List[VolumeTracing])(implicit mp: MessagesProvider, ctx: DBAccessContext): Fox[DataSet] =
+      volumeTracings: List[VolumeTracing],
+      wkUrl: String)(implicit mp: MessagesProvider, ctx: DBAccessContext): Fox[DataSet] =
     for {
       dataSetName <- assertAllOnSameDataSet(skeletonTracings, volumeTracings) ?~> "nml.file.differentDatasets"
       organizationNameOpt <- assertAllOnSameOrganization(skeletonTracings, volumeTracings) ?~> "nml.file.differentDatasets"
       organizationIdOpt <- Fox.runOptional(organizationNameOpt) {
         organizationDAO.findOneByName(_)(GlobalAccessContext).map(_._id)
-      } ?~> Messages("organization.notFound", organizationNameOpt.getOrElse("")) ~> NOT_FOUND
+      } ?~> (if (wkUrl.nonEmpty && conf.Http.uri != wkUrl) {
+               Messages("organization.notFound.wrongHost", organizationNameOpt.getOrElse(""), wkUrl, conf.Http.uri)
+             } else { Messages("organization.notFound", organizationNameOpt.getOrElse("")) }) ~>
+        NOT_FOUND
```

Comment:
Is this a hard error? As a developer, I occasionally download and upload NMLs from/to different instances. Could this be a warning only? propagation to the front-end is probably not straight-forward.. maybe the developer perspective is a bit too special here? cc @daniel-wer @fm3

Reply:
The wkurl is only used to augment the error message in case the organization or dataset name are not present (which already let the upload fail before this change). I hope this works for you

Reply:
ok, great :+1: